src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -86,6 +86,8 @@ class Docker implements Serializable {
                 def arg = parsedArgs[i]
                 if ((arg == '-f' || arg.startsWith('--file')) && i < (parsedArgs.length - 1)) {
                     dockerfile = arg.startsWith('--file=') ? arg.split('=')[1] : parsedArgs[i+1]
+                    // Trim leading and trailing quotes
+                    dockerfile = dockerfile.replaceAll('''^'|'$|^"|"$''', "")
```

Comment:
I think it would be a little clearer to use a `/` string. Also, do we need to make sure we don't trim mismatched quotes like `'foo"`? Maybe better to use something like this:

```suggestion
                    dockerfile = dockerfile.replaceAll(/^('|")(.*)(\1)$/, "$2")
```